src/main/java/com/dlsc/jfxcentral/data/model/Learn.java

```diff
@@ -0,0 +1,56 @@
+package com.dlsc.jfxcentral.data.model;
+
+import org.apache.commons.lang3.StringUtils;
+
+public class Learn extends ModelObject {
+
+    private String title;
+    private String sourceCode;
+    public Learn() {
+    }
+
+    @Override
+    public boolean matches(String searchPattern) {
+        if (super.matches(searchPattern)) {
+            return true;
+        }
+
+        if (StringUtils.containsAnyIgnoreCase(title, searchPattern)) {
+            return true;
+        }
+
+        return StringUtils.containsAnyIgnoreCase(sourceCode, searchPattern);
+    }
+
+    public String getTitle() {
+        return title;
+    }
+
+    public void setTitle(String title) {
+        this.title = title;
+    }
+
+    public String getSourceCode() {
+        return sourceCode;
+    }
+
+    public void setSourceCode(String sourceCode) {
```

Comment:
Why is there a field "sourceCode" on the model object? Isn't the code part of the markdown / the readme?

Reply:
also added Url, this is an optional link to a GitHub or GitLab project with the full code used in the article

for example:

```
  {
    "id": "led-strip",
    "createdOn": "2023-10-18",
    "modifiedOn": "2023-10-18",
    "name": "LED strip with a JavaFX UI",
    "personIds": [
      "f.delporte"
    ],
    "canonicalUrl": "https://pi4j.com/examples/jbang/pixelblaze_output_expander_javafx/",
    "sourceCodeUrl": "https://github.com/Pi4J/pi4j-jbang/PixelblazeOutputExpanderJavaFX.java"
  }
```